src/pip_sync_faster/main.py

```diff
@@ -23,13 +23,6 @@ def path(self):
     def abspath(self):
         return os.path.abspath(self._path)
 
-    @property
-    @lru_cache
-    def abspath_hash(self):
-        hash_ = hashlib.sha512()
-        hash_.update(self.abspath.encode("utf8"))
-        return hash_.hexdigest()
```

Comment:
It was using a hash of the absolute path to the requirements file as the cache key. There's no need: just use the absolute path directly as the cache key.